avro-codegen/src/main/java/com/linkedin/avroutil1/codegen/SpecificRecordClassGenerator.java

```diff
@@ -208,24 +209,23 @@ protected JavaFileObject generateSpecificRecord(AvroRecordSchema recordSchema, S
       classBuilder.addJavadoc(doc);
     }
 
-    if(config != null) {
-      if(config.getMinimumSupportedAvroVersion().laterThan(AvroVersion.AVRO_1_7)) {
-        // MODEL$ as SpecificData()
-        classBuilder.addField(
-            FieldSpec.builder(CLASSNAME_SPECIFIC_DATA, "MODEL$", Modifier.PRIVATE,
-                Modifier.STATIC)
-                .initializer(CodeBlock.of("new $T()", CLASSNAME_SPECIFIC_DATA))
-                .build());
-      } else {
-        classBuilder.addField(
-            FieldSpec.builder(CLASSNAME_SPECIFIC_DATA, "MODEL$", Modifier.PRIVATE,
-                Modifier.STATIC)
-                .initializer(CodeBlock.of("$T.get()", CLASSNAME_SPECIFIC_DATA))
-                .build());
-      }
-
+    if(config != null && config.getMinimumSupportedAvroVersion().laterThan(AvroVersion.AVRO_1_7)) {
```

Comment:
i think it will be better to substitute a null config with a default instance, or throw at the top for no config. would save you from a lot of null checks